pkg/uid/from_etc.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/coreos/rkt/pkg/group"
+	"github.com/coreos/rkt/pkg/passwd"
+	"github.com/hashicorp/errwrap"
+)
+
+// FromEtc is the struct that generates a uid/gid by parsing etc/passwd,
+// and etc/group relative from the given Root path looking for the given Name.
+type FromEtc struct {
```

Comment:
You missed a `NewFromEtc` for this one.

Further, since all of your constructors return the interface, there's no need to export any of these structs - and indeed, to me it's preferable not to, since it minimises the exposed interface to the package. However, obviously if you don't export them, you will need to test them within the same package (hi @tmrts).

And personally I think all the generators should be in a single file - it's a relatively trivial amount of code, and now uid_range.go is an odd exception in the package.

buuut - I want to leave the bulk of this review to @tmrts  ;-)